src/graph/InsertEdgeExecutor.cpp

```diff
@@ -47,51 +48,59 @@ void InsertEdgeExecutor::execute() {
         auto dst = row->dstid();
         auto rank = row->rank();
         auto expressions = row->values();
-        std::vector<VariantType> values;
+        std::vector<std::string> names;
+        std::vector<PropValue> values;
 
         values.reserve(expressions.size());
         for (auto *expr : expressions) {
-            values.emplace_back(expr->eval());
-        }
-
-        RowWriter writer(schema_);
-        for (auto &value : values) {
+            auto value = expr->eval();
+            PropValue propValue;
             switch (value.which()) {
                 case 0:
-                    writer << boost::get<int64_t>(value);
+                    propValue.set_int_val(boost::get<int64_t>(value));
+                    values.emplace_back(std::move(propValue));
                     break;
                 case 1:
-                    writer << boost::get<double>(value);
+                    propValue.set_double_val(boost::get<double>(value));
+                    values.emplace_back(std::move(propValue));
                     break;
                 case 2:
-                    writer << boost::get<bool>(value);
+                    propValue.set_bool_val(boost::get<bool>(value));
+                    values.emplace_back(std::move(propValue));
                     break;
                 case 3:
-                    writer << boost::get<std::string>(value);
+                    propValue.set_string_val(boost::get<std::string>(value));
+                    values.emplace_back(std::move(propValue));
                     break;
                 default:
                     LOG(FATAL) << "Unknown value type: " << static_cast<uint32_t>(value.which());
+                    onError_(Status::Error("Unknown value type"));
+                    return;
             }
         }
+
         {
             auto &out = edges[index++];
             out.key.set_src(src);
             out.key.set_dst(dst);
             out.key.set_ranking(rank);
             out.key.set_edge_type(edgeType_);
-            out.props = writer.encode();
+            for (auto property : properties_) {
```

Comment:
Serialize the property names for every row of values?

I seriously don't agree with that.

Reply:
In `Storage Layer`, I wish to check the fields is indexed or not .